galley/pkg/config/processing/runtime.go

```diff
@@ -0,0 +1,167 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package processing
+
+import (
+	"sync"
+	"sync/atomic"
+
+	"istio.io/api/mesh/v1alpha1"
+
+	"istio.io/istio/galley/pkg/config/event"
+)
+
+// ProcessorOptions are options that are passed to event.Processors during startup.
+type ProcessorOptions struct {
```

Comment:
Shouldn't `ProcessorOptions` be part of the definition for `Processor` rather than being in this file?

Reply:
Should the configuration for the processor be part of construction rather than a value passed in to `Start`? Passing in effectively a `void*` to Start seems a bit hacky.

Reply:
It needs to be part of Start call, as it defines the behavior between the Start/Stop calls.

Reply:
Is this due to the fact that processors can be restarted? How often do we expect restarts to occur? 

Providing a generic API that supports restarting adds a lot of complexity - I'm not sure it's worth it.  One option is to just re-create the processor each time a "restart" occurs ... this would push off the specialized arguments to a type-specific constructor method that takes `ProcessorOptions`.

Alternatively, you could use some sort of mix-in interface that defines a `StartWithOptions(ProcessorOptions)', to allow you to restart as you do now, but I'd avoid that if possible

Reply:
I considered recreate after each build when I was first writing this (which, admittedly is a long time ago now). The line-of-code cost of implementation is probably going to be more expensive (as we still want stop functionality) but will simplify the cognitive overhead for sure. I'll take another stab to see if a rebuild-for-restart model might simplify things.

Reply:
Changed the model to use a factory method for future stuff. Causes the code to be a bit more complicated. PTAL.